src/panels/config/application_credentials/dialog-add-application-credential.ts

```diff
@@ -0,0 +1,240 @@
+import "@material/mwc-button";
+import "@material/mwc-list/mwc-list-item";
+import {
+  css,
+  CSSResultGroup,
+  html,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+} from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { ComboBoxLitRenderer } from "lit-vaadin-helpers";
+import { fireEvent } from "../../../common/dom/fire_event";
+import "../../../components/ha-circular-progress";
+import "../../../components/ha-combo-box";
+import { createCloseHeading } from "../../../components/ha-dialog";
+import "../../../components/ha-textfield";
+import {
+  fetchApplicationCredentialsConfig,
+  createApplicationCredential,
+  ApplicationCredential,
+} from "../../../data/application_credential";
+import { domainToName } from "../../../data/integration";
+import { PolymerChangedEvent } from "../../../polymer-types";
+import { haStyleDialog } from "../../../resources/styles";
+import { HomeAssistant } from "../../../types";
+import { AddApplicationCredentialDialogParams } from "./show-dialog-add-application-credential";
+
+interface Domain {
+  id: string;
+  name: string;
+}
+
+const rowRenderer: ComboBoxLitRenderer<Domain> = (item) => html`<mwc-list-item>
+  <span>${item.name}</span>
+</mwc-list-item>`;
+
+@customElement("dialog-add-application-credential")
+export class DialogAddApplicationCredential extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _loading = false;
+
+  // Error message when can't talk to server etc
+  @state() private _error?: string;
+
+  @state() private _params?: AddApplicationCredentialDialogParams;
+
+  @state() private _domain?: string;
+
+  @state() private _clientId?: string;
+
+  @state() private _clientSecret?: string;
+
+  @state() private _domains?: Domain[];
+
+  public showDialog(params: AddApplicationCredentialDialogParams) {
+    this._params = params;
+    this._domain = "";
+    this._clientId = "";
+    this._clientSecret = "";
+    this._error = undefined;
+    this._loading = false;
+    this._fetchConfig();
+  }
+
+  protected firstUpdated(changedProperties: PropertyValues) {
+    super.firstUpdated(changedProperties);
+    this.addEventListener("keypress", (ev) => {
```

Comment:
I think this is handled by the dialog, and this is not needed?

Reply:
Removed.